lib/storage/src/content_manager/toc.rs

```diff
@@ -141,20 +128,15 @@ impl TableOfContent {
             segment_searcher: Box::new(SimpleCollectionSearcher::new()),
             collection_management_runtime,
             // PeerId does not change during execution so it is ok to copy it here.
-            #[cfg(feature = "consensus")]
             this_peer_id: raft_state.this_peer_id(),
-            #[cfg(feature = "consensus")]
             collection_meta_wal,
-            #[cfg(feature = "consensus")]
             raft_state: Arc::new(std::sync::Mutex::new(raft_state)),
-            #[cfg(feature = "consensus")]
             propose_sender: None,
-            #[cfg(feature = "consensus")]
             on_consensus_op_apply: std::sync::Mutex::new(HashMap::new()),
+            distributed_mode,
         }
     }
 
-    #[cfg(feature = "consensus")]
     pub fn with_propose_sender(&mut self, sender: std::sync::mpsc::Sender<Vec<u8>>) {
```

Comment:
As now we always have a propose sender, we can just pass it in `new` and remove an Option in ToC

Reply:
Good catch!
Here is a naive fix https://github.com/qdrant/qdrant/pull/565/commits/df56ed0728150b21ed5d0d32dbb83f2c695e438c

We could however use the `Option<propose_sender>` as a toggle switch for enabling the distributed mode.
Any preferences?

Reply:
@e-ivkov WDYT about https://github.com/qdrant/qdrant/pull/565/commits/029878f4670d20775b0749a412a1b8510f25438c